orttraining/orttraining/python/training/ortmodule/_custom_autograd_function_exporter.py

```diff
@@ -37,7 +37,7 @@ def _export_pt_1_10(g, n, *args, **kwargs):
                 "wrap exportable sub-nn.Module's as ORTModule."
             )
         inplace = kwargs["inplace"]
-        training_mode = symbolic_helper._training_mode
+        training_mode = symbolic_helper.check_training_mode
```

Comment:
`_training_mode` is a private field not exposed to users. Do you need access to the current training mode during export?

Reply:
+1

Reply:
Created https://github.com/pytorch/pytorch/issues/79950 cc @bowenbao

Reply:
For a temporary fix, id use torch.onnx._globals.GLOBAL.export_training` If there is anything to unblock.  https://github.com/pytorch/pytorch/blob/765b6a8fab02a55cf85058037f9ab89fb00a5b23/torch/onnx/_globals.py#L30 I do suggest leaving a todo and plan to update it once we expose the value properly. Do you have any suggestions on the api?

Reply:
For the release version of PyTorch 1.12, you _may_ need  torch.onnx._globals.GLOBAL.training_mode

Reply:
https://github.com/pytorch/pytorch/blob/734a97a7c828d6bfe82764c9cae399c349828091/torch/onnx/_globals.py#L28 From the 1.12 release branch: sounds right. Note that even though the field was marked as optional[…] it was really optional[bool]. The type was fixed after the 1.12 cut

Reply:
Is there anyway to know the training mode and keep it when calling export()? This way we don’t need to rely on an internal state

Reply:
> Is there anyway to know the training mode and keep it when calling export()? This way we don’t need to rely on an internal state

Have you looked into https://github.com/pytorch/pytorch/blob/master/torch/onnx/utils.py#L77-L124?

Basically the exporter tries to respect whatever `model.training` has, which all `torch.nn.Module` provides. The `torch.onnx.export(..., mode: _C_onnx.TrainingMode, ...)` is used to determine what a default value is in some cases.

Reply:
> `_training_mode` is a private field not exposed to users. Do you need access to the current training mode during export?
> 
> (check_training_mode is a function which returns None. It doesn't seem fitting to be used here)

One alternative is simply convert it to public/property :)

Reply:
I would like to be cautious about exposing internal states. Once we do that, changes will be very hard. I am hoping we remove more globals in the future.

Reply:
@justinchuby @ytaous 
FYI it's `torch.onnx._globals.GLOBALS.training_mode` not `torch.onnx._globals.GLOBAL.training_mode` 
I'll update this PR

Reply:
good catch sorry about that

Reply:
No worries. The PR is updated now